flocker/acceptance/obsolete/test_containers.py

```diff
@@ -335,3 +336,59 @@ def test_traffic_routed(self, cluster):
             # Connect to the machine where the container is NOT running:
             lambda _: assert_http_server(self, origin.public_address, port))
         return running
+
+    @require_cluster(2)
+    def test_reboot(self, cluster):
+        """
+        After a reboot the containers are only started once all datasets are
+        available locally.
+        """
+        # Find a node which is not running the control service.
+        # If the control node is rebooted, we won't get stale dataset state.
+        node = [node for node in cluster.nodes if
```

Comment:
:grey_question: Totally optional in this case, but for large lists this is slightly faster using a inline generator rather than a list. I came across a sort of cool construction for that on stackoverflow the other day and thought I'd share:

next(node for node in cluster.nodes if
        node.public_address != cluster.control_node.public_address)